weave/engine_trace.py

```diff
@@ -255,10 +261,38 @@ def flush_queue(self):
         self._orig_writer.flush_queue()
 
 
+def patch_ddtrace_set_tag():
+    from ddtrace import span as ddtrace_span
+    from inspect import signature
+
+    set_tag_signature = signature(ddtrace_span.Span.set_tag)
+
+    # replaces ddtrace.Span.set_tag and ddtrace.Span.set_metric if not patched already
+    if "redacted_val" not in set_tag_signature.parameters:
+        old_set_tag = ddtrace_span.Span.set_tag
+        old_set_metric = ddtrace_span.Span.set_metric
+
+        # Only logged redeacted values if flag is on
+        def set_tag(self, key, unredacted_val="", redacted_val=""):
+            old_set_tag(self, key, redacted_val) if os.getenv(
+                "DISABLE_WEAVE_PII"
+            ) else old_set_tag(self, key, unredacted_val)
+
+        # Dont log metrics if flag is on and not redacted
+        def set_metric(self, key, val="", is_pii_redacted=False):
+            old_set_metric(self, key, "") if os.getenv(
```

Comment:
this logic is still incorrect. if `is_pii_redacted` is True, then we should log the value

Reply:
oops - nvm - this is correct